state-chain/pallets/cf-flip/src/lib.rs

```diff
@@ -165,31 +227,46 @@ pub mod pallet {
 		AccountBonded,
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_current_block: BlockNumberFor<T>) -> Weight {
+			// Clear the call counter on every block. Do it in on_initialize (instead of
+			// `on_finalize`) so it's inspectable.
+			let _ = CallCounter::<T>::clear(u32::MAX, None);
+			T::WeightInfo::on_initialize()
+		}
+	}
+
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		/// Set the PER BLOCK slashing rate.
+		/// Apply a list of configuration updates to the pallet.
 		///
-		/// The dispatch origin of this function must be governance
-		///
-		/// ## Events
-		///
-		/// - [SlashingRateUpdated]
-		///
-		/// ## Errors
-		///
-		/// - [BadOrigin](frame_system::error::BadOrigin)
-		///
-		/// ## Dependencies
-		///
-		/// - [EnsureGovernance]
+		/// Requires Governance.
 		#[pallet::call_index(0)]
-		#[pallet::weight(T::WeightInfo::set_slashing_rate())]
-		pub fn set_slashing_rate(origin: OriginFor<T>, slashing_rate: Permill) -> DispatchResult {
+		#[pallet::weight(T::WeightInfo::update_pallet_config())]
+		pub fn update_pallet_config(
+			origin: OriginFor<T>,
+			updates: BoundedVec<PalletConfigUpdate, ConstU32<10>>,
+		) -> DispatchResult {
 			// Ensure the extrinsic was executed by the governance
 			T::EnsureGovernance::ensure_origin(origin)?;
-			// Set the slashing rate
-			SlashingRate::<T>::set(slashing_rate);
-			Self::deposit_event(Event::SlashingRateUpdated { slashing_rate });
+
+			for update in updates {
+				match update {
+					PalletConfigUpdate::SetSlashingRate(slashing_rate) => {
+						SlashingRate::<T>::set(slashing_rate);
+						Self::deposit_event(Event::SlashingRateUpdated { slashing_rate });
+					},
+					// You can used FixedU64::from_rational or from_float to convert the input
+					// number to FixedU64.
+					// The range is: [0.000000000, 18446744073.709551615]
+					// i.e. there are 9 decimal places.
+					PalletConfigUpdate::SetFeeScalingRate(fee_scaling_rate) => {
```

Comment:
do we want to set like a hardcoded "safety margin" for this scaling rate? e.g. 1 < scaling_rate < 100 or something?

Reply:
It's governance applied, so don't think it's necessary